README.md

```diff
@@ -31,6 +31,11 @@ export type StreamProps = {
    * Either the video id or the signed url for the video you’ve uploaded to Cloudflare Stream should be included here.
    */
   src: string;
+  /**
+   * Ref for accessing the underlying stream element. Useful for providing imperative access to the player API:
+   * https://developers.cloudflare.com/stream/viewing-videos/using-the-player-api
+   */
+  streamRef?: MutableRefObject<HTMLStreamElement | undefined>;
```

Comment:
Is it necessary to have the question mark and also | in undefined?

`MutableRefObject<HTMLStreamElement | undefined> | undefined` might be easier as `MutableRefObject<HTMLStreamElement> | undefined`?

Reply:
It is necessary here — the `| undefined` is inside the `MutableRefObject` generic type, which means the `.current` property on the ref might be `undefined`, whereas the `?` on `streamRef` makes `streamRef` optional as a prop.

Unfortunately, defining `streamRef`'s type as `MutableRefObject<HTMLStreamElement> | undefined` while omitting the `?` would have two unwanted effects:
1. It would mean the `.current` property is always guaranteed, which not only makes it uninstantiable for users,  but is incorrect since it doesn't get set until the first round of effects run inside our component
2. It would make `streamRef` required as a prop, even if the value is `undefined`
    e.g.
```tsx
return (
  <Stream src="xxxxxxxxxxxxxxx" streamRef={undefined} />
)
```